pkg/target/target_test.go

```diff
@@ -430,3 +438,239 @@ func TestProcessData(t *testing.T) {
 		})
 	}
 }
+
+func fooMatch() *match.Match {
+	return &match.Match{
+		Kinds: []match.Kinds{
+			{
+				Kinds:     []string{"Thing"},
+				APIGroups: []string{"some"},
+			},
+		},
+		Scope:              "Namespaced",
+		Namespaces:         []util.Wildcard{"my-ns"},
+		ExcludedNamespaces: nil,
+		LabelSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"obj": "label",
+			},
+		},
+		NamespaceSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"ns": "label",
+			},
+		},
+		Name: "",
+	}
+}
+
+func fooConstraint() *unstructured.Unstructured {
+	return makeConstraint(
+		setKinds([]string{"some"}, []string{"Thing"}),
+		setScope("Namespaced"),
+		setNamespaceName("my-ns"),
+		setLabelSelector("obj", "label"),
+		setNamespaceSelector("ns", "label"),
+	)
+}
+
+func TestToMatcher(t *testing.T) {
+	unstructAssign, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(&unversioned.Assign{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: unversioned.AssignSpec{
+			Match: *fooMatch(),
+		},
+	})
+	tests := []struct {
+		name       string
+		constraint *unstructured.Unstructured
+		want       constraints.Matcher
+		wantErr    error
+	}{
+		{
+			name:       "constraint with no match fields",
+			constraint: makeConstraint(),
+			wantErr:    ErrCreatingMather,
+		},
+		{
+			name:       "constraint with match fields",
+			constraint: fooConstraint(),
+			want: &Matcher{
+				fooMatch(),
+			},
+		},
+		{
+			name: "mutator with match fields",
+			constraint: &unstructured.Unstructured{
+				Object: unstructAssign,
+			},
+			want: &Matcher{
+				fooMatch(),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := &K8sValidationTarget{}
+			got, err := h.ToMatcher(tt.constraint)
+			if !errors.Is(err, tt.wantErr) {
+				t.Errorf("ToMatcher() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if diff := cmp.Diff(got, tt.want, cmp.AllowUnexported(Matcher{})); diff != "" {
```

Comment:
Since the only member of Matcher{} is an unexported member, this diff will always return `""`. Instead, compare the underlying matcher to ensure it makes sense. So

```
cmp.Diff(tt.want.match, got.match)
```

The fields on `match.Match` are exported, so cmp.Diff will work properly on them.

Reply:
Per discussion, I was wrong. I thought this was "IgnoreUnexported"